sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java

```diff
@@ -161,120 +162,238 @@ public static void showCaseUpdateWithNewDiseaseVariantDialog(CaseDataDto existin
+		// b) Tested disease != case disease: Ask user to create a new case for the tested disease
+
 		CaseDataDto preSaveCaseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(associatedCase.getUuid());
 		CaseDataDto postSaveCaseDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(associatedCase.getUuid());
 		showSaveNotification(preSaveCaseDto, postSaveCaseDto);
 
-		Runnable confirmCaseCallback = () -> {
-			if (dto.getTestedDisease() == postSaveCaseDto.getDisease()
-				&& PathogenTestResultType.POSITIVE.equals(dto.getTestResult())
-				&& dto.getTestResultVerified()
-				&& postSaveCaseDto.getCaseClassification() != CaseClassification.CONFIRMED
-				&& postSaveCaseDto.getCaseClassification() != CaseClassification.NO_CASE) {
-				showConfirmCaseDialog(postSaveCaseDto);
-			}
-		};
+		final boolean equalDisease = dto.getTestedDisease() == postSaveCaseDto.getDisease();
 
-		Runnable caseCloningCallback = () -> {
-			if (dto.getTestedDisease() != postSaveCaseDto.getDisease()
-				&& dto.getTestResult() == PathogenTestResultType.POSITIVE
-				&& dto.getTestResultVerified()) {
-				showCaseCloningWithNewDiseaseDialog(postSaveCaseDto, dto.getTestedDisease());
+		Runnable negativeTestCallback = () -> {
+			if (equalDisease && PathogenTestResultType.NEGATIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
+				changeAssociatedSampleResult(dto, handleChanges -> {
+				});
 			}
 		};
 
-		Runnable caseDiseaseVariantCallback = () -> {
-			if (dto.getTestedDiseaseVariant() != null
-				&& !DataHelper.equal(dto.getTestedDiseaseVariant(), postSaveCaseDto.getDiseaseVariant())
-				&& dto.getTestResult() == PathogenTestResultType.POSITIVE
-				&& dto.getTestResultVerified()) {
-				showCaseUpdateWithNewDiseaseVariantDialog(postSaveCaseDto, dto.getTestedDiseaseVariant());
+		Runnable positiveTestCallback = () -> {
+			if (PathogenTestResultType.POSITIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
+				if (equalDisease) {
+					changeAssociatedSampleResult(dto, handleChanges -> {
+						if (handleChanges) {
+							if (dto.getTestedDiseaseVariant() != null
+								&& !DataHelper.equal(dto.getTestedDiseaseVariant(), postSaveCaseDto.getDiseaseVariant())) {
+								showCaseUpdateWithNewDiseaseVariantDialog(
+									postSaveCaseDto,
+									dto.getTestedDiseaseVariant(),
+									() -> {
+										// Retrieve the case again because it might have changed
+										showConfirmCaseDialog(FacadeProvider.getCaseFacade().getByUuid(postSaveCaseDto.getUuid()));
+									});
+							} else {
+								showConfirmCaseDialog(postSaveCaseDto);
+							}
+						}
+					});
+				} else {
+					showCaseCloningWithNewDiseaseDialog(postSaveCaseDto, dto.getTestedDisease());
+				}
 			}
 		};
 
 		if (onSavedPathogenTest != null) {
 			onSavedPathogenTest.accept(dto, () -> {
-				confirmCaseCallback.run();
-				caseCloningCallback.run();
-				caseDiseaseVariantCallback.run();
+				positiveTestCallback.run();
```

Comment:
Technically, there is no need to provide two separate callbacks here for positive and negative test results.
However, it is okay if it is supposed to be more readable like this.